### PR TITLE
fix(electrostatics): correct multipole background term

### LINKED ISSUE #156: 🐛[BUG]: Multipole PME has wrong background energy
### Version

0.4.1

### On which installation method(s) does this occur?

_No response_

### Describe the issue

For a charged system, `multipole_particle_mesh_ewald` predicts an incorrect background energy correction, too large by a factor of $4\pi$, where the expected term is $E_{bg} = -k_e \pi Q^2 / (2 \alpha^2 V)$. The cause is `multipole_pme_energy_corrections` using `FIELD_CONSTANT` ($= 4 \pi k_e$) where the $k=0$ term needs plain $k_e$.

In addition:

- `multipole_electrostatic_energy` doesn't include the background energy, while it is supposed to be mathematically identical to `multipole_ewald_summation` according to the documentation.
- It is quite inconsistent to include the background correction in `multipole_pme_reciprocal_space` and not in `multipole_reciprocal_space_energy`.

### Minimum reproducible example

```python
import math

import numpy as np
import torch

from nvalchemiops.torch.interactions.electrostatics import (
    multipole_electrostatic_energy,
    multipole_ewald_summation,
    multipole_particle_mesh_ewald,
    pack_multipole_moments,
)
from nvalchemiops.torch.neighbors import neighbor_list

device = torch.device("cuda" if torch.cuda.is_available() else "cpu")

KE = 14.399645351950548               # e^2/(4 pi eps0), eV.A/e^2
L, SIGMA, ALPHA = 8.0, 1.0, 1.0       # cell, GTO width, Ewald damping
R_CUT, K_CUTOFF, MESH = 10.0, 12.0, (64, 64, 64)
VOL = L ** 3


def system(qtot=0.0):
    """2x2x2 rock salt with total charge `qtot`."""
    frac = [(p + np.array([i, j, k])) / 2.0
            for i in range(2) for j in range(2) for k in range(2)
            for p in ([0.0, 0.0, 0.0], [0.5, 0.5, 0.5])]
    pos = torch.tensor(np.array(frac) * L, dtype=torch.float64, device=device)
    q = torch.tensor([1.0, -1.0] * 8, dtype=torch.float64, device=device)
    return pos, q + (qtot - q.sum()) / q.shape[0]


POS, _ = system()
CELL = torch.eye(3, dtype=torch.float64, device=device) * L
PBC = torch.tensor([[True, True, True]], device=device)
NL, PTR, SHIFTS = neighbor_list(
    POS, cutoff=R_CUT, cell=CELL.unsqueeze(0), pbc=PBC, return_neighbor_list=True,
)
IDX_J = NL[1].contiguous()


def direct(q):
    return float(multipole_electrostatic_energy(
        POS, pack_multipole_moments(q), CELL,
        sigma=SIGMA, k_cutoff=K_CUTOFF, include_self_interaction=False,
    ).sum())


def ewald(q):
    return float(multipole_ewald_summation(
        POS, pack_multipole_moments(q), CELL, IDX_J, PTR, SHIFTS,
        sigma=SIGMA, alpha=ALPHA, k_cutoff=K_CUTOFF,
    ).sum())


def pme(q):
    return float(multipole_particle_mesh_ewald(
        POS, pack_multipole_moments(q), CELL, IDX_J, PTR, SHIFTS,
        sigma=SIGMA, alpha=ALPHA, mesh_dimensions=MESH,
    ).sum())


print(f"  {'Q':>4}  {'direct':>15} {'ewald_summation':>16} {'pme':>16}"
      f"   {'(e-d)/bg':>9} {'(p-d)/bg':>9} {'(p-e)/bg':>9}")
for qtot in (0.0, 2.0, 4.0, 8.0):
    _, q = system(qtot)
    d, e, p = direct(q), ewald(q), pme(q)
    if qtot == 0.0:
        print(f"  {qtot:4.1f}  {d:15.9f} {e:16.9f} {p:16.9f}")
        continue
    bg = -KE * math.pi * qtot ** 2 / (2 * ALPHA ** 2 * VOL)
    print(f"  {qtot:4.1f}  {d:15.9f} {e:16.9f} {p:16.9f}"
          f"   {(e - d) / bg:9.4f} {(p - d) / bg:9.4f} {(p - e) / bg:9.4f}")
```

### Relevant log output

```shell
Q           direct  ewald_summation              pme    (e-d)/bg  (p-d)/bg  (p-e)/bg
   0.0    -55.633847296    -55.633826135    -55.633805099
   2.0    -56.636979250    -56.460248237    -58.680833470     -1.0001   11.5661   12.5663
   4.0    -59.646375111    -58.939514542    -67.821918582     -1.0000   11.5663   12.5663
   8.0    -71.683958555    -68.856579762   -104.386259032     -1.0000   11.5664   12.5664
```

### Environment details

```shell
GPU: NVIDIA H200
CUDA version: 13.3
Python version: 3.11.6
PyTorch version: 2.11.0
NVIDIA Warp version: 1.16.0
```

### PR DESCRIPTION
**Pull request by zubatyuk**
# ALCHEMI Toolkit-Ops Pull Request

## Description

Fix the multipole uniform-background correction for non-neutral periodic systems. Use the standard `FIELD_CONSTANT / (8 alpha^2 V)` coefficient and subtract the correction consistently in direct-sum Ewald, PME, and cached Ewald calculations while retaining the direct reciprocal zero-mode convention.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [x] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Related Issues

Fixes #156

## Changes Made

- Correct the multipole PME background coefficient and centralize its per-atom, single-system, and batched evaluation.
- Apply the same correction to direct-sum and cached Ewald energies for monopoles, dipoles, and quadrupoles.
- Extend existing charged-system coverage for cross-method agreement, alpha invariance, batching, charge and volume derivatives, and update the documented convention.

## Testing

- [x] Unit tests pass locally (`make pytest`)
- [x] Linting passes (`make lint`)
- [x] New tests added for new functionality meets coverage expectations?

## Checklist

- [x] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [x] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [x] I have performed a self-review of my code
- [x] I have added docstrings to new functions/classes
- [x] I have updated the documentation (if applicable)

## Additional Notes

None.

> [!TIP]
> This repository uses Greptile, an AI code review service, to help conduct
> pull request reviews. We encourage contributors to read and consider suggestions
> made by Greptile, but note that human maintainers will provide the necessary
> reviews for merging: Greptile's comments are **not** a qualitative judgement
> of your code, nor is it an indication that the PR will be accepted/rejected.
> We encourage the use of emoji reactions to Greptile comments, depending on
> their usefulness and accuracy.

### CODE REVIEW

File: nvalchemiops/torch/interactions/electrostatics/pme_multipole.py [hunk 1]
```diff
@@ -4269,6 +4269,50 @@ def _corrections_double_backward_fake(  # pragma: no cover
...
+    ).scatter_add(0, batch_idx, charges_f64)
+    vol_per_system = volume.to(torch.float64).reshape(-1)
+    if vol_per_system.numel() == 1 and n_systems > 1:
+        vol_per_system = vol_per_system.expand(n_systems)
+    return (
+        c_bg_no_v
+        * charges_f64
+        * total_charge.index_select(0, batch_idx)
+        / vol_per_system.index_select(0, batch_idx)
+    )
```
[the remark this answers is not in the document]
Comment (zubatyuk) [reply nvalchemiops/torch/interactions/electrostatics/pme_multipole.py:4313]: False, in `torch>=2.8` `scatter_add` and `index_select` accept int32.

**Comment by laserkelvin** [comment]
/ok to test [618ee78](https://github.com/NVIDIA/nvalchemi-toolkit-ops/pull/163/commits/618ee78690da1fe4379458ff98f54ef924dc3cd1)

File: nvalchemiops/torch/interactions/electrostatics/multipole_ewald.py [hunk 2]
```diff
@@ -4897,6 +4897,32 @@ def _coeff(value: float) -> torch.Tensor:
...
+    alpha: float,
+    volume: torch.Tensor,
+    *,
+    batch_idx: torch.Tensor | None = None,
+    n_systems: int | None = None,
+) -> torch.Tensor:
+    """Return the positive per-atom uniform-background correction.
+
+    This is subtracted from the split Ewald energy so it matches the direct-k
+    convention with the reciprocal zero mode removed.
```
[the remark this answers is not in the document]
Comment (zubatyuk) [reply nvalchemiops/torch/interactions/electrostatics/multipole_ewald.py]: Fixed in 0146530

**Review by dallasfoster — APPROVED** [review]

**Comment by dallasfoster** [comment]
/ok to test [e6ca7bd](https://github.com/NVIDIA/nvalchemi-toolkit-ops/pull/163/commits/e6ca7bd972e58de77f3eb14d77ce36beb09b6e3f)
